### PR TITLE
Factorize base engine run into one function

### PR DESCRIPTION
**Pull request by norbusan**
To unify treatment of multiple runs necessary between LatexConverter and PDFLatexConverter, factor the run loop into a separate function that is called from generate_pdf.

### CODE REVIEW

**Review by ntai-arxiv — APPROVED** [review]
TBH, I dislike the use of function ref. If you need to inject dependency, there should be a better way than using a variable of function ref.

Why not name _latex_run and _pdflatex_run as _base_run? 

Anyhow, it is too hard for me to read this in browser, and also want to study with a bit of time so merge this, and I'd like to make some surface level changes.

**Review by ntai-arxiv — APPROVED** [review]

File: tex2pdf_service/tex2pdf/tex_to_pdf_converters.py [hunk 1]
```diff
@@ -78,6 +78,52 @@ def produce_pdf(self, tex_file: str, work_dir: str, in_dir: str, out_dir: str) -
         """Produce PDF from the given tex file. Return the outcome dict."""
         pass
 
+    @abstractmethod
+    def _base_runner(self, step: str, tex_file:str, work_dir: str, in_dir: str, out_dir: str) -> dict:
```
Comment (ntai-arxiv) [inline tex2pdf_service/tex2pdf/tex_to_pdf_converters.py]: suggest _latexen_run in lie of _base_runner, and change _lapex_run and _pdflatex_run to be _latexen_run.

And get rid of _base_runner overriding.

**Review by ntai-arxiv — COMMENTED** [review]
Thanks for the refactor the ugly part. I dislike the use of `_base_runner = _latex_run`. It seems more straightforward use it as override functions

**Review by ntai-arxiv — APPROVED** [review]
I changed _base_runner part.

**Comment by norbusan** [comment]
> I changed _base_runner part.

Great, thanks for that!!! I was contemplating a similar change, but again - wasn't sure about the original intention of naming the functions, so I kept them.

The way it is now is much cleaner, thanks!

**Review by ntai-arxiv — APPROVED** [review]

**Pull request merged**